src/components/timeline/GlobalTrack.js

```diff
@@ -0,0 +1,279 @@
+      // If this global track is hidden, render out a stub element so that the
+      // Reorderable Component still works across all the tracks.
+      return <li className="timelineTrackHidden" />;
+    }
+
+    return (
+      <li className="timelineTrack" style={style}>
+        <div
+          className={classNames('timelineTrackRow timelineTrackGlobalRow', {
+            selected: isSelected,
+          })}
+          onClick={this._onLineClick}
+        >
+          <ContextMenuTrigger
+            id={'TimelineTrackContextMenu'}
+            renderTag="div"
+            attributes={{
+              title: titleText,
+              className: 'timelineTrackLabel timelineTrackGlobalGrippy',
+              onMouseDown: this._onLabelMouseDown,
+            }}
+          >
+            <h1 className="timelineTrackName">{trackName}</h1>
+          </ContextMenuTrigger>
+          <div className="timelineTrackTrack">{this.renderTrack()}</div>
+        </div>
+        <Reorderable
+          tagName="ol"
+          className="timelineTrackLocalTracks"
+          order={localTrackOrder}
+          orient="vertical"
+          grippyClassName="timelineTrackLocalGrippy"
+          onChangeOrder={this._changeLocalTrackOrder}
+        >
+          {localTracks.map((localTrack, trackIndex) => {
+            if (pid === null) {
+              console.error(
+                'The pid should never be null when adding a TimelineLocalTrack.',
+                pid
+              );
+              return <div />;
+            }
+            return (
+              <TimelineLocalTrack
+                key={trackIndex}
+                pid={pid}
+                localTrack={localTrack}
+                trackIndex={trackIndex}
+              />
+            );
+          })}
+        </Reorderable>
+      </li>
+    );
+  }
+}
+
+function getGlobalTrackName(
+  globalTrack: GlobalTrack,
+  threads: Thread[]
+): string {
+  switch (globalTrack.type) {
+    case 'process': {
+      // Look up the thread information for the process if it exists.
+      return globalTrack.mainThreadIndex === null
+        ? `Process ${globalTrack.pid}`
+        : getFriendlyThreadName(threads, threads[globalTrack.mainThreadIndex]);
+    }
+    case 'screenshots':
+      return 'Screenshots';
+    default:
+      throw new Error(`Unhandled GlobalTrack type ${(globalTrack: empty)}`);
+  }
+}
+
+// Provide some empty lists, so that strict equality checks work for component updates.
+const EMPTY_TRACK_ORDER = [];
+const EMPTY_LOCAL_TRACKS = [];
+
+const options: ExplicitConnectOptions<OwnProps, StateProps, DispatchProps> = {
+  mapStateToProps: (state, { trackIndex }) => {
+    const globalTracks = getGlobalTracks(state);
+    const globalTrack = globalTracks[trackIndex];
+
+    // These get assigned based on the track type.
+    let threadIndex = null;
+    let isSelected = false;
+    let titleText = null;
+    let trackName = getGlobalTrackName(globalTrack, getThreads(state));
```

Comment:
I'm a bit concerned about this call: it calls `getFriendlyThreadName` that's a bit time-consuming. Given that this function is called for each store change, it should be as lean as possible and use only memoized selectors.

The possible solutions that I see are:
* compute the track name in the component's constructor + componentWillReceiveProps and put it in the state.
* move this somehow to a selector.
* do this computation in a `mergeProps` function (but this would likely require changing our `explicitConnect` typing): `mapStateToProps` would return only memoized things, and `areStatePropsEqual` would be able to run and decide to skip rendering before `mergeProps` even runs.
* memoize the local `getGlobalTrackName` right here.

Reply:
Nice point! Let's do a selector that returns an array of all the names, that way we compute them all once, then don't have to worry about parameterizing a memoized selector.